api/src/backend/api/v1/views.py

```diff
@@ -2565,3 +2572,130 @@ def get_serializer_context(self):
+        """
+        instance = self.get_object()
+        if not instance.api_key_decoded:
+            return Response(
+                {"detail": "API key is invalid or missing."},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        try:
+            client = openai.OpenAI(
+                api_key=instance.api_key_decoded,
+            )
+            models = client.models.list()
+            return Response(
+                {
+                    "detail": "Connection successful!",
+                    "available_models": [model.id for model in models.data],
+                },
+                status=status.HTTP_200_OK,
+            )
+        except Exception as e:
+            return Response(
+                {"detail": f"Connection failed: {str(e)}"},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+    def create(self, request, *args, **kwargs):
+        """Create new AI configuration"""
+        if LighthouseConfig.objects.filter(tenant_id=request.tenant_id).exists():
+            return Response(
+                {"detail": "AI configuration already exists. Use PUT to update."},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        instance = serializer.save()
+
+        headers = self.get_success_headers(serializer.data)
+        return Response(
+            LighthouseConfigSerializer(
+                instance, context=self.get_serializer_context()
+            ).data,
+            status=status.HTTP_201_CREATED,
+            headers=headers,
+        )
+
+    def update(self, request, *args, **kwargs):
+        """Update existing AI configuration"""
+        instance = self.get_object()
+        serializer = self.get_serializer(instance, data=request.data)
+        serializer.is_valid(raise_exception=True)
+        instance = serializer.save()
+        return Response(
+            LighthouseConfigSerializer(
+                instance, context=self.get_serializer_context()
+            ).data
+        )
+
+    @action(detail=True, methods=["get"], url_path="show_key")
+    def show_key(self, request, pk=None):
+        """
+        Return the decrypted API key for the specified AI configuration.
+        """
+        instance = self.get_object()
+
+        try:
+            # Handle different binary formats appropriately
+            if isinstance(instance.api_key, memoryview):
+                encrypted_bytes = instance.api_key.tobytes()
+            elif isinstance(instance.api_key, str):
+                encrypted_bytes = instance.api_key.encode()
+            else:
+                encrypted_bytes = instance.api_key
+
+            key = settings.SECRETS_ENCRYPTION_KEY.encode()
+            fernet_instance = Fernet(key)
+            decrypted_key = fernet_instance.decrypt(encrypted_bytes).decode()
```

Comment:
I don't understand how there can be different formats for the key that we store, could you explain why this can happen?

Reply:
This is sorted using following code:

```
        try:
            api_key_bytes = bytes(self.api_key)
            decrypted_key = fernet.decrypt(api_key_bytes)
            return decrypted_key.decode()
```